index.ts

```diff
@@ -32,35 +32,39 @@ function axiosToJson(data: any) {
   return data;
 }
 
-export const bind: runtime.client.ClientAdapter = async (
-  arg: runtime.server.EndpointArg<any, any, any, any>
-): Promise<runtime.server.Response<any, any, any, Record<string, any>>> => {
-  if (arg.servers.length !== 1) {
-    return assert.fail('cannot decide which server to use from ' + arg.servers.join(', '));
-  }
-  const server = arg.servers[0];
-  const params = axiosToJson(arg.query);
-  const data = toRequestData(arg.body);
-  const url = server + arg.path;
-  const headers = { ...arg.headers, ...(data instanceof FormData ? data.getHeaders() : {}) };
-  const response = await axios.request({
-    method: arg.method,
-    headers,
-    url,
-    params,
-    data,
-    validateStatus: () => true
-  });
-  const contentType = getContentType(response);
-  return {
-    status: response.status,
-    value: {
-      contentType,
-      value: getResponseData(contentType, response)
-    },
-    headers: response.headers ?? {}
+export const bind: runtime.client.ClientAdapter = withAxios(globalAxios);
+
+export function withAxios(axiosInstance: AxiosInstance): runtime.client.ClientAdapter {
+  return async (
+    arg: runtime.server.EndpointArg<any, any, any, any>
+  ): Promise<runtime.server.Response<any, any, any, Record<string, any>>> => {
+    if (arg.servers.length !== 1) {
+      return assert.fail('cannot decide which server to use from ' + arg.servers.join(', '));
+    }
+    const server = arg.servers[0];
```

Comment:
So I think here we need to check if `axiosInstance.defaults.baseURL` is set and then use that.

Reply:
Maybe it is just fine to have this be specified explicitly when providing the adapter. So let's leave it like this.